src/widget/wtracktableview.cpp

```diff
@@ -421,28 +421,17 @@ void WTrackTableView::assignPreviousTrackColor() {
         ColorPaletteSettings colorPaletteSettings(m_pConfig);
         ColorPalette colorPalette = colorPaletteSettings.getTrackColorPalette();
         mixxx::RgbColor::optional_t color = pTrack->getColor();
-        pTrack->setColor(colorPalette.previousColor(color));
-    }
-}
 
-void WTrackTableView::assignNextTrackColor() {
-    QModelIndexList indices = getSelectedRows();
-    if (indices.isEmpty()) {
-        return;
-    }
-
-    TrackModel* trackModel = getTrackModel();
-    if (!trackModel) {
-        return;
-    }
-
-    QModelIndex index = indices.at(0);
-    TrackPointer pTrack = trackModel->getTrack(index);
-    if (pTrack) {
-        ColorPaletteSettings colorPaletteSettings(m_pConfig);
-        ColorPalette colorPalette = colorPaletteSettings.getTrackColorPalette();
-        mixxx::RgbColor::optional_t color = pTrack->getColor();
-        pTrack->setColor(colorPalette.nextColor(color));
+        while (steps != 0) {
+            if (steps > 0) {
+                color = colorPalette.nextColor(color);
+                steps--;
+            } else {
+                color = colorPalette.previousColor(color);
+                steps++;
+            }
+        }
```

Comment:
this solution still does a lot of searching. we can go one step further 
```suggestion
        const int currentColorIdx = colorPalette.indexOf(color);
        const int newColorIndex = rem_euclid(currentColorIdx + steps, colorPalette.size());
        color = colorPalette.at(newColorIndex);
```
`rem_euclid` is a euclidean remainder/modulo which you'll have to implement in util/math.h`.

You might also want to consider to implement this as `nextNthColor` in `ColorPalette` directly.

Reply:
Good suggestion !
But i'm not sure the modulo efficiency is relevant here

Reply:
Here probably not, no, but if we add it as a general function to math, we should still make sure its within reasonable limits.

Reply:
What about simply moving this to colorpalette so it is available elsewhere?
`ColorPalette::getNthColor()`

Reply:
I simply moved the while loop the getNthColor, as suggested by @ronso0 because there are too many individual cases that are already handled in nextColor and previousColor (for example index -1 is no color, and you need to cycle through it too).

I felt that trying to be overefficient here would damage readability and not gain execution time given the nature of the color change (only manual changes of a few steps can happen).

I called this function in the per deck track_color_selector too.

Reply:
Test results:
The steps maximum value is 99, min is -99. (is it hardcoded in the control definition ?)
It takes 0ms to execute and there are no visual artifacts.
I made sure we really cycle through the palette, even with big values.
Works for Library and Channel.

Reply:
yup, the visual artifact problem should be solved, 0ms doesn't mean its very efficient though. At least add a TODO at the loop location please.